browser/ui/views/brave_help_bubble/brave_help_bubble_view.cc

```diff
@@ -0,0 +1,209 @@
+                              base::Unretained(this)));
+}
+
+void BraveHelpBubbleView::AddedToWidget() {
+  InitElementTrackers();
+
+  if (tracked_element_ && tracked_element_->GetVisible()) {
+    UpdatePosition();
+    Show();
+  }
+}
+
+void BraveHelpBubbleView::UpdatePosition() {
+  auto tracked_element_origin =
+      tracked_element_->GetBoundsInScreen().CenterPoint();
+  tracked_element_origin.set_x(tracked_element_origin.x() - (kWidth / 2) - 1.f);
+  tracked_element_origin.set_y(tracked_element_origin.y() - kHeight - 15);
+
+  SetPosition({tracked_element_origin.x(), tracked_element_origin.y()});
+}
+
+void BraveHelpBubbleView::Show() {
+  if (!brave_help_bubble_delegate_) {
+    brave_help_bubble_delegate_ = new BraveHelpBubbleDelegate(this, text_);
+    brave_help_bubble_delegate_->AddObserver(this);
+  }
+
+  if (brave_help_bubble_delegate_) {
+    brave_help_bubble_delegate_->Show();
+  }
+
+  SetPaintToLayer();
+  layer()->SetFillsBoundsOpaquely(false);
+  SetSize({kWidth, kHeight});
+  SchedulePulsingAnimation(layer());
+  SetVisible(true);
+}
+
+void BraveHelpBubbleView::Hide() {
+  if (brave_help_bubble_delegate_) {
+    brave_help_bubble_delegate_->Hide();
+  }
+
+  SetVisible(false);
+}
+
+void BraveHelpBubbleView::OnBubbleClosing(Widget* widget) {
+  // In the destruction, we don't have to remove this from parent. Otherwise,
+  // DCHECK will fail as the ancestor view is iterating its children to destroy
+  // all descendant.
+  if (GetWidget()->IsClosed())
+    return;
+
+  // There's a possibility that someone is iterating over the parent's children,
+  // and we are removing the child view early. This can cause a DCHECK in
+  // view.cc, so we schedule removal of the child view via TaskRunner to avoid
+  // this issue.
+  base::SequencedTaskRunner::GetCurrentDefault()->PostTask(
+      FROM_HERE,
+      base::BindOnce(
+          [](base::WeakPtr<views::View> view, View* this_element) {
+            if (!view)
+              return;
+            if (!view->parent() || !view->parent()->Contains(view.get()))
+              return;
+            view->parent()->RemoveChildView(this_element);
+          },
+          weak_factory_.GetWeakPtr(), this));
+
+  brave_help_bubble_delegate_ = nullptr;
+  base::SequencedTaskRunner::GetCurrentDefault()->DeleteSoon(FROM_HERE, this);
+}
+
+void BraveHelpBubbleView::OnPaint(gfx::Canvas* canvas) {
+  cc::PaintFlags flags;
+  flags.setColor(kBlockColor);
+  flags.setAntiAlias(true);
+  flags.setStyle(cc::PaintFlags::kStroke_Style);
+  flags.setShader(kBraveGradient);
+  flags.setStrokeWidth(2.f);
+  canvas->DrawCircle(GetContentsBounds().CenterPoint(), 27.f, flags);
+  flags.setStrokeWidth(5.f);
+  canvas->DrawCircle(GetContentsBounds().CenterPoint(), 20.f, flags);
+}
+
+void BraveHelpBubbleView::OnTrackedElementShown(ui::TrackedElement* element) {
+  auto* el = element->AsA<views::TrackedElementViews>();
+
+  if (!el || el->view() != tracked_element_)
+    return;
```

Comment:
totally optional: we can collpse these into

The benefit from this collapsing is that we can narrow down the `el`'s scope.

```suggestion
  if (auto* el = element->AsA<views::TrackedElementViews>(); !el || el->view() != tracked_element_) {
      return;
  }
```

Reply:
This is a clever usage of eager evaluation. I like it.